src/Microsoft.SqlTools.ServiceLayer/Rename/Requests/ProcessRenameEditRequest.cs

```diff
@@ -0,0 +1,22 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+using Microsoft.SqlTools.Hosting.Protocol.Contracts;
+using Microsoft.SqlTools.Utility;
+
+namespace Microsoft.SqlTools.ServiceLayer.Rename.Requests
+{
+    /// <summary>
+    /// Property class for Rename Service
+    /// </summary>
+    public class ProcessRenameEditRequestParams : GeneralRequestDetails
+    {
+        public RenameTableInfo TableInfo { get; set; }
```

Comment:
let's make this generic enough: `ObjectPath` or something similar that you can pass in from azure data studio side .

Reply:
and of course, to begin with, you can choose to only implement the table and column, but the whole thing should be very easy to extend.

Reply:
That is absolutely true. As @Charles-Gagnon [pointed out](https://github.com/microsoft/sqltoolsservice/pull/1686#discussion_r972304258), I change the whole class to get the Urn of the object and the new name. It makes the whole operation much simpler.